### DISCUSSION TITLE
Using OpenTofu State File Encryption with Terragrunt

### DISCUSSION THREAD

**Discussion by NIXKnight** [gruntwork-io/knowledge-base#853] [How do I]
I'm exploring the integration of OpenTofu's state encryption capabilities with Terragrunt and was wondering if anyone has experience or examples demonstrating how to set this up effectively.

Specifically, I am looking for an example configuration that utilizes OpenTofu's state encryption in a Terragrunt-managed setup.

If there’s a public repository, blog post, or guide that covers this, I would greatly appreciate a pointer. Otherwise, any direct advice or snippets would be incredibly helpful!

**Comment by NIXKnight** [comment]
One way is to use the Terragrunt [generate](https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#generate) block.

**Comment by yhakbar** [comment]
We're working with the community to get it supported natively:
https://github.com/gruntwork-io/terragrunt/pull/3586
